modules/carousel/Carousel.tsx

```diff
@@ -222,79 +248,106 @@ const Carousel: React.FC<ICarouselProps & WithStylesProps<Theme, ICarouselStyles
+          const childrenSize = calculateChildrenSize();
 
-        const numberOfFullyVisibleCards = Math.floor(scrollElementWidth / childrenSize);
-        const lastCardVisibleWidth = scrollElementWidth - childrenSize * numberOfFullyVisibleCards;
+          const numberOfFullyVisibleCards = Math.floor(scrollElementWidth / childrenSize);
+          const lastCardVisibleWidth =
+            scrollElementWidth - childrenSize * numberOfFullyVisibleCards;
 
-        let newOffset = 0;
-        if (rightDir) {
-          newOffset = childrenSize - lastCardVisibleWidth - halfMarginBetweenItems;
+          let newOffset = 0;
+          if (rightDir) {
+            newOffset = childrenSize - lastCardVisibleWidth - halfMarginBetweenItems;
 
-          while (currentScroll >= Math.floor(newOffset)) {
-            newOffset += childrenSize;
-          }
+            while (currentScroll >= Math.floor(newOffset)) {
+              newOffset += childrenSize;
+            }
 
-          /**
-           * This 0.3 value means that if we don't see only 30% or less of the
-           * last element it jumps to the next one
-           */
-          if (
-            Math.abs(currentScroll - newOffset) <
-            lastCardVisibleWidth * notVisiblePercentageToSkipElement
-          ) {
-            newOffset += childrenSize;
-          }
+            /**
+             * This 0.3 value means that if we don't see only 30% or less of the
+             * last element it jumps to the next one
+             */
+            if (
+              Math.abs(currentScroll - newOffset) <
+              lastCardVisibleWidth * notVisiblePercentageToSkipElement
+            ) {
+              newOffset += childrenSize;
+            }
 
-          newOffset -= halfMarginBetweenItems;
-        } else {
-          while (currentScroll > Math.floor(newOffset)) {
-            newOffset += childrenSize;
+            newOffset -= halfMarginBetweenItems;
+          } else {
+            while (currentScroll > Math.floor(newOffset)) {
+              newOffset += childrenSize;
+            }
+            newOffset = newOffset < childrenSize ? 0 : newOffset - childrenSize;
+          }
+          element.scrollTo(newOffset, 0);
+        } else if (scrollType === "centered") {
+          if (rightDir) {
+            setCenteredItemInd(centeredItemInd + 1);
+            element.scrollTo(element.scrollLeft + calculateChildrenSize(), 0);
+          } else {
+            setCenteredItemInd(centeredItemInd - 1);
+            element.scrollTo(element.scrollLeft - calculateChildrenSize(), 0);
           }
-          newOffset = newOffset < childrenSize ? 0 : newOffset - childrenSize;
         }
-        element.scrollTo(newOffset, 0);
       }
     },
     [
       calculateChildrenSize,
+      centeredItemInd,
       marginBetweenItems,
       notVisiblePercentageToSkipElement,
+      scrollType,
       scrollableElementRef
     ]
   );
 
+  useEffect(() => {
```

Comment:
A comment here explaining that this useEffect initially centers the element with `centeredItemInd` would be nice! 🙂